src/words/environment/WordsObject.java

```diff
@@ -13,13 +15,35 @@
 	private Position cell;
 	private String currentMessage;
 	private Action lastAction;
-
+	private HashMap<WordsObject, ArrayList<Property>> referers;
```

Comment:
Oh interesting, yea I just kind of automatically go to ArrayList as my default iterable collection. I guess it could be a LinkedList. Something about iterating through a Set doesn't sit well with me. I feel like it's not meant to be used for that. I don't know I could be wrong. But yea let's leave it.